script/prerender/manage.ts

```diff
@@ -0,0 +1,397 @@
+    const numSuccesses = successfulMessages.length;
+    if (numSuccesses < pageBatch.length) {
+      throw new Error(`[SQS] [SendMessageBatch] Unexpected response: received only ${numSuccesses
+        } successes out of ${pageBatch.length} pages in a batch but also no failures. Successes: ${
+        JSON.stringify(successfulMessages)}`);
+    }
+  }
+
+  console.log(`[${book.title}] All ${numBookPages} book pages queued`);
+
+  // console.log(`[${book.title}] Queuing sitemap`);
+
+  await sqsClient.send(new SendMessageCommand({
+    MessageBody: JSON.stringify({ payload: { pages, slug: book.slug }, type: 'sitemap' }),
+    QueueUrl: workQueueUrl,
+  }));
+
+  console.log(`[${book.title}] Sitemap queued`);
+
+  // Used in the sitemap index
+  return {
+    params: { book: { slug: book.slug } },
+    state: { bookUid: book.id, bookVersion: book.version },
+  };
+}
+
+async function manage() {
+  console.log('Preloading route components');
+
+  await Loadable.preloadAll();
+
+  archiveLoader = createArchiveLoader(REACT_APP_ARCHIVE_URL, {
+    appPrefix: '',
+    archivePrefix: ARCHIVE_URL,
+  });
+  osWebLoader = createOSWebLoader(`${OS_WEB_URL}${REACT_APP_OS_WEB_API_URL}`);
+
+  bookLoader = makeUnifiedBookLoader(archiveLoader, osWebLoader);
+
+  // We can start fetching book ToCs while the stack is created,
+  // but we need a limit so we don't use up all the memory
+  console.log(`Loading books in batches of ${MAX_CONCURRENT_BOOKS}`);
+
+  const allPagesQueuedPromise = asyncPool(
+    MAX_CONCURRENT_BOOKS, Object.entries(BOOKS), prepareAndQueueBook
+  );
+
+  // Wait for the workers stack to be ready
+  await waitUntilStackCreateComplete(
+    {client: cfnClient, maxWaitTime: WORKERS_DEPLOY_TIMEOUT_SECONDS, minDelay: 10, maxDelay: 10},
+    {StackName: WORKERS_STACK_NAME}
+  );
+
+  console.log('Retrieving queue URLs');
+
+  // Get the queue URLs
+  const describeStacksResult = await cfnClient.send(
+    new DescribeStacksCommand({StackName: WORKERS_STACK_NAME})
+  );
+
+  const stacks = assertDefined(
+    describeStacksResult.Stacks, '[CFN] [DescribeStacks] Unexpected response: missing Stacks key'
+  );
+  const stack = assertDefined(stacks[0], `${WORKERS_STACK_NAME} stack not found`);
+  const outputs = assertDefined(
+    stack.Outputs, '[CFN] [DescribeStacks] Unexpected response: missing stack Outputs key'
+  );
+
+  let workQueueUrl: string | undefined;
+  let deadLetterQueueUrl: string | undefined;
+
+  for (const output of outputs) {
+    const outputKey = assertDefined(
+      output.OutputKey,
+      '[CFN] [DescribeStacks] Unexpected response: missing output OutputKey key'
+    );
+    const outputValue = assertDefined(
+      output.OutputValue,
+      '[CFN] [DescribeStacks] Unexpected response: missing output OutputValue key'
+    );
+
+    switch (outputKey) {
+      case 'WorkQueueUrl':
+        workQueueUrl = outputValue;
+        resolveWorkQueuePromise(workQueueUrl);
```

Comment:
0 seconds since the workers take some time to boot even after the stack is deployed and the manager can start queuing up some messages before they are ready. :) Fine, I'll rewrite this.

Reply:
Done, it now waits for the stack creation at the beginning and gets the queue urls before it starts loading any of the books.